src/uri/params.cr

```diff
@@ -362,6 +362,44 @@ class URI
       raw_params.delete(name)
     end
 
+    # Merges *params* into self.
+    #
+    # ```
+    # params = URI::Params.parse("foo=bar&foo=baz&qux=zoo")
+    # other_params = URI::Params.parse("foo=buzz&foo=extra")
+    # params.merge!(other_params).to_s # => "foo=buzz&foo=extra&qux=zoo"
+    # params.fetch_all("foo")          # => ["buzz", "extra"]
+    # ```
+    #
+    # See `#merge` for a non-mutating alternative
+    def merge!(params : URI::Params, *, replace : Bool = true) : URI::Params
+      if replace
+        @raw_params.merge!(params.raw_params) { |_, _, second| second.dup }
```

Comment:
Following #13474, this will currently need different names for the unused parameters:
```suggestion
        @raw_params.merge!(params.raw_params) { |_, _first, second| second.dup }
```